AsyncDisplayKit/Layout/ASDimension.mm

```diff
@@ -17,32 +17,35 @@
 
 ASOVERLOADABLE ASDimension ASDimensionMake(NSString *dimension)
 {
-  // Handle empty string
-  if (dimension.length == 0) {
-    return ASDimensionMake(ASDimensionUnitPoints, 0.0);
-  }
-  
-  // Handle points
-  NSUInteger pointsStringLocation = [dimension rangeOfString:@"pt"].location;
-  if (pointsStringLocation != NSNotFound) {
-    // Check if points is at the end and remove it
-    if (pointsStringLocation == (dimension.length-2)) {
-      dimension = [dimension substringToIndex:(dimension.length-2)];
-      return ASDimensionMake(ASDimensionUnitPoints, dimension.floatValue);
+  if (dimension.length != 0) {
+    
+    // Handle points
+    NSUInteger pointsStringLocation = [dimension rangeOfString:@"pt"].location;
```

Comment:
One tricky way to perform this check would be characterAtIndex: length - 1, and then check if it == 't' or == '%' — and if it is neither, then return Auto?  Probably too obtuse, just having fun :)